### PR TITLE
SR2019 floorplans

### PR DESCRIPTION
**Pull request by WillB97**
Initial pull request of Student Robotics 2018 competition floorplans for setup and public use.
Contains floorplan templates and an SVG generation script to produce all required diagrams through hiding layers and merging template images.
Opened to allow collaboration on designs, see srobo/tasks#64 and  srobo/tasks#65.

### CODE REVIEW

**Comment by PeterJCLaw** [comment]
@WillB97 minor meta point: this year's competition is the "SR2019" competition -- our naming scheme is to use the calendar year in which the competition event happens.

File: 2018/README.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
+# Notes for 2018
+
+The templates here are intended to be 1:1 scale diagram of all the areas used in SUSU, they are believed to be within 5% accurate.
+Locations highlighted in red on one of the venue related layers require measurements to be taken to assure their size.
+
+#### Requires:
+- \> python 3.4
```
Comment (PeterJCLaw) [inline 2018/README.md]: The `generate_svg.py` script uses f-strings, which are in Python 3.6+ only. Should the f-strings be removed, or is this requirement wrong?

File: 2018/README.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+#### Folder structure:
+```
+team_names.txt
+templates/
+  Lv4.svg
+  Lv3.svg
+  Lv2.svg
+  Cube.svg
+  key.svg
+  layout.svg
```
Comment (PeterJCLaw) [inline 2018/README.md]: `layout.svg` seems to be missing and the other names don't seem to match the names of the files in the `templates` directory. Is this expected?
This seems to be an issue as the `venue_map.yaml` file expects the `layout.svg` template to exist.

(meta: I'd be tempted to drop this sort of detail from the REAMDE, including specific file names here is a recipe for them getting out of date).

**Comment by PeterJCLaw** [comment]
I'm getting a syntax error trying to run this:
```
$ python3.6 ./generate_svg.py l2_public.yaml
Using yaml
Traceback (most recent call last):
  File "/usr/lib/python3.6/xml/etree/ElementPath.py", line 263, in iterfind
    selector = _cache[cache_key]
KeyError: ('.//svg:text[svg:tspan="{{title}}" ]', (('', 'http://www.w3.org/2000/svg'), ('cc', 'http://creativecommons.org/ns#'), ('dc', 'http://purl.org/dc/elements/1.1/'), ('inkscape', 'http://www.inkscape.org/namespaces/inkscape'), ('rdf', 'http://www.w3.org/1999/02/22-rdf-syntax-ns#'), ('sodipodi', 'http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd'), ('svg', 'http://www.w3.org/2000/svg')))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./generate_svg.py", line 60, in <module>
    root.find('.//svg:text[svg:tspan="{{title}}" ]', ns)[0].text = spec.get('title',spec['image'])
  File "/usr/lib/python3.6/xml/etree/ElementPath.py", line 298, in find
    return next(iterfind(elem, path, namespaces), None)
  File "/usr/lib/python3.6/xml/etree/ElementPath.py", line 277, in iterfind
    selector.append(ops[token[0]](next, token))
  File "/usr/lib/python3.6/xml/etree/ElementPath.py", line 233, in prepare_predicate
    raise SyntaxError("invalid predicate")
SyntaxError: invalid predicate
```

File: 2018/README.md (hunk 1 above)
Comment (WillB97) [reply 2018/README.md]: This is an error, it should be 3.6

**Comment by WillB97** [comment]
I have been unable to reproduce your issue, if it is still occuring could you print the contents of `ns` on the preceding line.

**Comment by RealOrangeOne** [comment]
Overall, this looks great! Commenting on an SVG like this is quite difficult, so i'm going to have to do it here (tick off once complete):
- [x] https://github.com/srobo/comp-floorplans/pull/3#issuecomment-468920724
- [ ] Can we get a barrier of some sort down the middle of the reception desks, which will only be used during check-in. It would be useful to ferry people towards 1 of the desks. Alternatively, if possible, an additional set of tables so we can have 4 rows of people checking in.
- [x] Files are stored in `2018
- [ ] Can we get a barrier between bar 3 and the TV studio? Would be useful so it's obvious competitors etc shouldn't go there.
- [x] I notice there's very little open are teams can use to test their robots in outside the arena. Newbury used the whole of floor 1. Any suggestions where teams might be able to use? Theoretically the actual arena and their pits are enough, but they're going to ask, so we might as well have an answer prepared!
- [ ] Is it worth putting a barrier on the stage to make sure people don't walk into / tamper with the battery charging?
- [ ] Charging all those batteries off just 2 wall sockets sounds like it could be _interesting_, we may need to pull the power from the other sockets on the stage.
- [ ] If we have scope for it, having a display in the volunteer area could be quite nice.

**Comment by trickeydan** [comment]
>     * [ ]  Charging all those batteries off just 2 wall sockets sounds like it could be _interesting_, we may need to pull the power from the other sockets on the stage.

I hear rumours from @Tyler-Ward of a 32A socket nearby that location.

**Comment by WillB97** [comment]
> * [x]  [#3 (comment)](https://github.com/srobo/comp-floorplans/pull/3#issuecomment-468920724)
> * [x]  Files are stored in `2018

My bad, I thought I'd already corrected that

> * [ ]  I notice there's very little open area teams can use to test their robots in outside the arena. Newbury used the whole of floor 1. Any suggestions where teams might be able to use? Theoretically the actual arena and their pits are enough, but they're going to ask, so we might as well have an answer prepared!

The seating in Bar 3 can be moved so we could move a few bays to make an area for testing.

> * [ ]  If we have scope for it, having a display in the volunteer area could be quite nice.

Assuming there is a network socket in the room this shouldn't be to difficult, however SUSU only have 3 screens on wheels so we will have to decide where they are best used.

> I hear rumours from @Tyler-Ward of a 32A socket nearby that location.

There is a 32A socket in the corner of the cube so we need to make sure there is sufficient distro to break that down into multiple 13A sockets

**Comment by WillB97** [comment]
> * [x]  Can we get a barrier of some sort down the middle of the reception desks, which will only be used during check-in. It would be useful to ferry people towards 1 of the desks. Alternatively, if possible, an additional set of tables so we can have 4 rows of people checking in.
> * [x]  Can we get a barrier between bar 3 and the TV studio? Would be useful so it's obvious competitors etc shouldn't go there.
> * [x]  Is it worth putting a barrier on the stage to make sure people don't walk into / tamper with the battery charging?

I believe SUSU has rope barriers although this is worth checking in https://github.com/srobo/tasks/issues/67

**Comment by Tyler-Ward** [comment]
With the charging power, according to my maths even if both the charger and charger PSU are only 50% efficient we can still theoretically run 32 chargers of a 13A socket (not that that is a good idea) but the two next to charging should be fine if we do 2 4ways with 4 4ways in each.

For score screens in Bar3 and the volunteer area we can use computer displays rather than large TVs if we are short that way we save the good screens for the main areas.

some extra minor fixes.
- [ ] add a large display screen inside the cube.
- [ ] add score display on bar 3 seating area

**Comment by WillB97** [comment]
> With the charging power, according to my maths even if both the charger and charger PSU are only 50% efficient we can still theoretically run 32 chargers of a 13A socket (not that that is a good idea) but the two next to charging should be fine if we do 2 4ways with 4 4ways in each.

Just because you can doesn't mean you should.
 
> * [ ]  add a large display screen inside the cube.

Where abouts in the cube? Beside staging or up on the stage?

**Comment by Tyler-Ward** [comment]
The screen in staging is a small screen for shepherding and staging uses, the big one will want to go by the doors and this shows the same score display as the ones in the team areas

**Review by Tyler-Ward — APPROVED** [review]
These are looking good to me. lets get these merged in.

**Review by mildlyincompetent — APPROVED** [review]
Looks good, we may want to make minor changes later but this is a good base.

**Comment by PeterJCLaw** [comment]
@WillB97 thanks for preparing these. (I've finally found time and managed to work out how to view them)

Would it be possible to change the colour of the squares used to indicate displays? It's somewhat confusing that they're the same colour as fire exits.

**Comment by PeterJCLaw** [comment]
> I have been unable to reproduce your issue, if it is still occuring could you print the contents of `ns` on the preceding line.

```
$ git diff
diff --git a/2019/generate_svg.py b/2019/generate_svg.py
index 54176ef..89d3035 100755
--- a/2019/generate_svg.py
+++ b/2019/generate_svg.py
@@ -56,6 +56,8 @@ root_tree = ET.parse('templates/' + spec['image']) # load svg
 root = root_tree.getroot()
 
 
+from pprint import pprint
+pprint(ns)
 try: # set title and version
   root.find('.//svg:text[svg:tspan="{{title}}"]', ns)[0].text = spec.get('title',spec['image'])
   root.find('.//svg:text[svg:tspan="{{version}}"]', ns)[0].text = "Version: " + str(spec.get('version',0.1))
```
yields
```
{'': 'http://www.w3.org/2000/svg',
 'cc': 'http://creativecommons.org/ns#',
 'dc': 'http://purl.org/dc/elements/1.1/',
 'inkscape': 'http://www.inkscape.org/namespaces/inkscape',
 'osb': 'http://www.openswatchbook.org/uri/2009/osb',
 'rdf': 'http://www.w3.org/1999/02/22-rdf-syntax-ns#',
 'sodipodi': 'http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd',
 'svg': 'http://www.w3.org/2000/svg'}
```

**Pull request merged**